src/NexusMods.App.UI/RightContent/Downloads/InProgressViewModel.cs

```diff
@@ -1,53 +1,263 @@
+                    x => new DownloadStatusView()
+                    {
+                        ViewModel = new DownloadStatusViewModel() { Row = x }
+                    }, DownloadColumn.DownloadStatus));
+        });
+
+        this.WhenActivated(d =>
+        {
+            TaskSourceChangeSet.Bind(out _tasksObservable)
+                .Subscribe()
+                .DisposeWith(d);
+
+            columns.Connect()
+                .Bind(out _filteredColumns)
                 .Subscribe()
                 .DisposeWith(d);
 
+            SuspendSelectedTasksCommand = ReactiveCommand.Create(
+                    () => { SuspendTasks(SelectedTasks.Items); },
+                    SelectedTasks.Connect()
+                        .AutoRefresh(task => task.Status)
+                        .Select(_ => SelectedTasks.Items.Any(task => task.Status == DownloadTaskStatus.Downloading)))
+                .DisposeWith(d);
+
+            ResumeSelectedTasksCommand = ReactiveCommand.Create(
+                    () => { ResumeTasks(SelectedTasks.Items); },
+                    SelectedTasks.Connect()
+                        .AutoRefresh(task => task.Status)
+                        .Select(_ => SelectedTasks.Items.Any(task => task.Status == DownloadTaskStatus.Paused)))
+                .DisposeWith(d);
+
+            SuspendAllTasksCommand = ReactiveCommand.Create(
+                () => { SuspendTasks(Tasks); },
+                Tasks.ToObservableChangeSet()
+                    .AutoRefresh(task => task.Status)
+                    .Select(_ => Tasks.Any(task => task.Status == DownloadTaskStatus.Downloading)));
+
+            ResumeAllTasksCommand = ReactiveCommand.Create(
+                () => { ResumeTasks(Tasks); },
+                Tasks.ToObservableChangeSet()
+                    .AutoRefresh(task => task.Status)
+                    .Select(_ => Tasks.Any(task => task.Status == DownloadTaskStatus.Paused)));
+
+            Tasks.ToObservableChangeSet()
+                .AutoRefresh(task => task.Status)
+                .Subscribe(_ =>
+                {
+                    UpdateWindowInfo();
+                    ActiveDownloadCount = Tasks.Count(task => task.Status == DownloadTaskStatus.Downloading);
+                    HasDownloads = Tasks.Any();
+                });
+
+            // Start updating on the UI thread
+            // This is a service to provide polling in case of downloaders that don't have a way of notifying
+            // changes in progress (e.g. when file name is resolved, or download progress is made).
+
+            // This is not the only mechanism, a manual refresh also can occur when tasks are added/removed.
+            var timer = new DispatcherTimer();
+            timer.Interval = TimeSpan.FromMilliseconds(PollTimeMilliseconds);
+            timer.Tick += UpdateWindowInfoEvent;
+            timer.Start();
+            Disposable.Create(timer, (tmr) => tmr.Stop()).DisposeWith(d);
```

Comment:
Is there a benefit to using `DispatcherTimer` instead of an observable stream?

Reply:
This is older code, I think from @Sewer56 . If there are no objection I'll change it to an observable stream for consistency.